core/tools/objectinspector/objectinspector.cpp

```diff
@@ -206,3 +210,61 @@ void ObjectInspector::scanForConnectionIssues()
         }
     }
 }
+
+void ObjectInspector::scanForThreadAffinityIssues()
+{
+    const auto &objects = Probe::instance()->allQObjects();
+
+    QMutexLocker lock(Probe::objectLock());
+    for (const auto object : objects) {
+        if (!Probe::instance()->isValidObject(object)) {
+            continue;
+        }
+
+        const auto objectName = Util::displayString(object);
+        if (object == object->thread()) {
+            const auto objectName = Util::displayString(object);
+            Problem problem;
+            problem.severity = Problem::Warning;
+            problem.description = QStringLiteral("The thread %1 has affinity with itself.").arg(objectName);
+            problem.object = ObjectId(object);
+            // p.location = bindingNode->sourceLocation(); //TODO can we get source locations of new-statements?
```

Comment:
See Probe::objectCreationSourceLocation().

Reply:
Thanks for the tip. I just realised that the `Problem` class uses a `QVector<SourceLocation>`, while it would have been nice to use `Execution::Trace`.
Is there any reason for that?

For tis PR I will just use `problem.locations.append(probe->objectCreationSourceLocation(object));`

Reply:
The idea of QVector<SourceLocation> is not to hold a trace, but to hold different relevant locations. For a duplicated connect issue, you'd want to see the locations of both connects, not just the second (I'm not sure, this is actually implemented yet, but that's why locations is a vector).
Using `problem.locations.append(probe->objectCreationSourceLocation(object));` is exactly how it's meant to be used :)

Reply:
OK, makes sense, and for the backtrace, there's always the "Show in object tool".